ReverseSequence.lua

```diff
@@ -0,0 +1,69 @@
+------------------------------------------------------------------------
+--[[ ReverseSequence ]] --
+-- Reverses a sequence on a given dimension.
+-- Example: Given a tensor of torch.Tensor({{1,2,3,4,5}, {6,7,8,9,10})
+-- nn.ReverseSequence(1):forward(tensor) would give: torch.Tensor({{6,7,8,9,10},{1,2,3,4,5}})
+------------------------------------------------------------------------
+local ReverseSequence, parent = torch.class("nn.ReverseSequence", "nn.Module")
+
+function ReverseSequence:__init(dim)
+    parent.__init(self)
+    self.output = torch.Tensor()
+    self.gradInput = torch.Tensor()
+    assert(dim, "Must specify dimension to reverse sequence over")
+    assert(dim <= 3, "Dimension has to be no greater than 3 (Only supports up to a 3D Tensor).")
+    self.dim = dim
+end
+
+function ReverseSequence:reverseOutput(input)
+    self.output:resize(input:size()):zero()
+    -- reverse output
+    local k = 1
+    for i = input:size(1), 1, -1 do
+        self.output[k] = input[i]
```

Comment:
I'm worried that an explicit loop for reversing will be very slow for CudaTensors, since each execution of this line will require device synchronization and a kernel launch. Can you reverse the sequence in a vectorized way to avoid this overhead?